### PR TITLE
Fix detecting release versions

### PR DESCRIPTION
**Pull request by ldziedziul**
Fix to update `hazelcast` and `hazelcast-x.y` brew packages only for release versions

Added unit tests for some bash scripts and run them within the CI workflow: 
image

For unit tests, I've used https://github.com/torokmark/assert.sh as it's simple and small enough to put it directly into our repository

### CODE REVIEW

File: build-hazelcast-homebrew-package.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/usr/local/bin/bash
+#!/usr/bin/env bash
```
[the remark this answers is not in the document]
Comment (frant-hartm) [reply build-hazelcast-homebrew-package.sh:1]: Or if it works with the mac version of bash we can remove the `brew install bash`.

File: packages/brew/test_brew_functions.sh [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  isReleaseVersion "$version"
+  assert_eq 0 "$?" "Version $version should be a release version" || TESTS_RESULT=$?
+}
+
+log_header "Tests for isReleaseVersion"
+assertNotReleaseVersion "5.2-SNAPSHOT"
+assertNotReleaseVersion "5.2-BETA-1"
+assertNotReleaseVersion "5.1-DR8"
+assertReleaseVersion "5.0"
+assertReleaseVersion "5.1"
```
Comment (frant-hartm) [inline packages/brew/test_brew_functions.sh:48]:
```suggestion
assertReleaseVersion "5.1"
assertReleaseVersion "5.1.1"
```

File: packages/brew/test_brew_functions.sh [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  local testValue=$1
+  local expected=$2
+  local actual=$(alphanumCamelCase "$testValue")
+  assert_eq "$expected" "$actual" "Alphanumeric camel case of $testValue should be equal to $expected " || TESTS_RESULT=$?
+}
+
+log_header "Tests for alphanumCamelCase"
+assertAlphanumCamelCase "5.2-SNAPSHOT" "52SNAPSHOT"
+assertAlphanumCamelCase "5.2-BETA-1" "52BETA1"
+assertAlphanumCamelCase "5.1-DR8" "51DR8"
```
[the remark this answers is not in the document]
Comment (ldziedziul) [reply packages/brew/test_brew_functions.sh]: Silly me, good catch! Fixed: https://github.com/hazelcast/hazelcast-packaging/pull/128/commits/20f0677f551aa68841ad50028323221394d6ea5f

File: build-hazelcast-homebrew-package.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (ldziedziul) [reply build-hazelcast-homebrew-package.sh:1]: On my machine it picks up the mac version of bash:
```
$ /usr/bin/env bash --version
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin21)
Copyright (C) 2007 Free Software Foundation, Inc.

$ /usr/local/bin/bash --version
GNU bash, version 5.1.8(1)-release (x86_64-apple-darwin20.3.0)
Copyright (C) 2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
```

but it works fine. I've removed brew bash: https://github.com/hazelcast/hazelcast-packaging/pull/128/commits/e75f0d9f600ec5a5de41511377451df11489b775

**Review by frant-hartm — APPROVED** [review]

**Pull request merged**